### PR TITLE
Updated Overall Attendance

### PR DESCRIPTION
**Pull request by ezhil56x**
## Description

Earlier the overallAttendance was calculated by averaging the attendance of all courses. Now it have been changed to weighted average to accurately calculate the overallAttendance.

### CODE REVIEW

**Comment by therealsujitk** [comment]
Thanks for contributing!

File: app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java [hunk 1]
```diff
@@ -1318,16 +1320,15 @@ private void downloadAttendance() {
                     attendanceItem.total = this.getIntegerValue(attendanceObject, "total");
 
                     if (attendanceItem.attended != null && attendanceItem.total != null && attendanceItem.total != 0) {
-                        attendanceItem.percentage = (int) Math.ceil((attendanceItem.attended * 100.0) / attendanceItem.total);
```
Comment (therealsujitk) [inline app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java:1321]: Why did you remove `attendanceItem.percentage`? This is required to display the percentage in the courses fragment.

File: app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java [hunk 2]
```diff
@@ -1300,7 +1300,9 @@ private void downloadAttendance() {
                 List<Attendance> attendance = new ArrayList<>();
 
                 float overallAttendance = 0;
```
Comment (therealsujitk) [inline app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java:1302]: Since `overallAttendance` is stored as an `int` anyway, can you change it to `int` here.

File: app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java [hunk 3]
```diff
@@ -1318,16 +1320,15 @@ private void downloadAttendance() {
...
                     attendance.add(attendanceItem);
                 }
 
-                if (attendanceLength != 0) {
-                    overallAttendance /= attendanceLength;
+                if (totalClasses != 0) {
+                    overallAttendance = (attendedClasses * 100.0f) / totalClasses;
                 }
 
                 sharedPreferences.edit().putInt("overallAttendance", (int) overallAttendance).apply();
```
Comment (therealsujitk) [inline app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java:1334]: Consequently this can just be `100` instead of `100.0f`, and the type casting can be removed.

File: app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java (hunk 1 above)
Comment (ezhil56x) [reply app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java:1321]: Done 👍🏻

File: app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java (hunk 2 above)
Comment (ezhil56x) [reply app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java]: Done 👍🏻

File: app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java (hunk 3 above)
Comment (ezhil56x) [reply app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java]: Done 👍🏻

File: app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java [hunk 4]
```diff
@@ -1319,15 +1321,15 @@ private void downloadAttendance() {
...
+                        totalClasses += attendanceItem.total;
                     }
 
                     attendance.add(attendanceItem);
                 }
 
-                if (attendanceLength != 0) {
-                    overallAttendance /= attendanceLength;
+                if (totalClasses != 0) {
+                    overallAttendance = (int) ((attendedClasses * 100) / totalClasses);
```
Comment (therealsujitk) [inline app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java:1332]: This type casting is redundant, make it.

```java
overallAttendance = (attendedClasses * 100) / totalClasses;
```

Comment (therealsujitk) [reply app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java:1332]: Also move the `overallAttendance` declaration to just above this `if` statement, since it isn't used anywhere before.

Comment (ezhil56x) [reply app/src/main/java/tk/therealsuji/vtopchennai/services/VTOPService.java]: Done 👍🏻

**Review by therealsujitk — APPROVED** [review]
